config/config/src/sources/manual.rs

```diff
@@ -0,0 +1,94 @@
+//! Manual source
+//!
+//! A manual source lets you set values manually.
+//!
+//! ```
+//! # use config::{sources, Value};
+//! #
+//! # #[derive(config::Config, serde::Deserialize)]
+//! # struct MyConfig {
+//! #     // ...
+//! # }
+//! #
+//! # fn main() -> Result<(), config::ConfigError> {
+//! let mut builder = config::ConfigBuilder::new();
+//! // Create a new ManualSource
+//! let mut manual = sources::ManualSource::new();
+//! manual.set("test.foo", Value::Bool(true));
+//! // Add ManualSource
+//! builder.add_source(manual);
+//! // Build the final configuration
+//! let config: MyConfig = builder.build()?;
+//! # Ok(())
+//! # }
+//! ```
+
+use std::{collections::BTreeMap, marker::PhantomData};
+
+use crate::{Config, ErrorSource, KeyPath, KeyPathSegment, Source, Value};
+
+use super::utils;
+
+/// Manual source
+///
+/// Create a new manual source with [`ManualSource::new`](ManualSource::new).
+pub struct ManualSource<C: Config> {
+    value: Option<Value>,
+    _phantom: PhantomData<C>,
+}
+
+fn value_to_value_graph(path: KeyPath, value: Value) -> Option<Value> {
+    match path.clone().into_iter().next() {
+        Some(segment) => {
+            let new = value_to_value_graph(path.drop_root(), value.clone());
+            match segment {
+                KeyPathSegment::Map { key } => {
+                    let mut map = BTreeMap::new();
+                    map.insert(key, new.unwrap_or(value));
+                    Some(Value::Map(map))
+                }
+                KeyPathSegment::Seq { .. } => unimplemented!(),
+            }
+        }
+        None => None,
+    }
+}
+
+impl<C: Config> Default for ManualSource<C> {
+    fn default() -> Self {
+        Self {
+            value: None,
+            _phantom: PhantomData,
+        }
+    }
+}
+
+impl<C: Config> ManualSource<C> {
+    /// Creates a new manual source.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Sets a value at the given path.
+    pub fn set<K: Into<KeyPath>>(&mut self, path: K, value: Value) {
```

Comment:
you need to get the `KeyGraph` for the key in `C` and then call the transform function on said graph.